SpatialGDK/Source/SpatialGDKFunctionalTests/SpatialGDK/SpatialTestRepNotify/SpatialTestRepNotify.cpp

```diff
@@ -256,11 +265,65 @@ void ASpatialTestRepNotify::PrepareTest()
 			FinishStep();
 		},
 		nullptr, 5.0f);
+
+	/**
+	 * In this section of the test, we check the order rep notifies are called in.
+	 * In native, data will be applied for an actor, its subobjects, and only THEN will repnotifies be called for the updated data.
+	 *
+	 * To test this, we set the update a replicated property on an actor (the test itself) and a subobject.
+	 * Then in repnotifies on both the actor and the subobject, we check that the property on the other object has already been updated.
+	 * */
+
+	AddStep(
+		TEXT("SpatialTestRepNotifyAllWorkersInitialiseExpectedOrderingProps"), FWorkerDefinition::AllWorkers, nullptr,
+		[this]() {
+			AssertIsValid(TestSubobject, TEXT("TestSubobject should be valid."));
+			TestSubobject->ExpectedParentInt1Property = 350;
+			TestSubobject->bParentPropertyWasExpectedProperty = false;
+			ExpectedSubobjectIntProperty = 400;
+			bSubobjectIntPropertyWasExpectedProperty = false;
+			FinishStep();
+		},
+		nullptr);
+
+	AddStep(
+		TEXT("SpatialTestRepNotifyServerChangeReplicatedVariables2"), FWorkerDefinition::Server(1), nullptr,
+		[this]() {
+			AssertIsValid(TestSubobject, TEXT("TestSubobject should be valid."));
+			TestSubobject->OnChangedRepNotifyInt = 400;
+			OnChangedRepNotifyInt1 = 350;
+			FinishStep();
+		},
+		nullptr);
+
+	AddStep(
+		TEXT("SpatialTestRepNotifyClientCheckOrderingWasCorrect"), FWorkerDefinition::AllClients,
+		[this]() -> bool {
+			return TestSubobject->OnChangedRepNotifyInt == 400 && OnChangedRepNotifyInt1 == 350;
+		},
+		[this]() {
+			AssertIsValid(TestSubobject, TEXT("TestSubobject should be valid."));
+			AssertEqual_Bool(TestSubobject->bParentPropertyWasExpectedProperty, true,
+							 TEXT("The OnChangedRepNotifyInt1 on parent actor ASpatialTestRepNotify should have been set to 350 before the "
+								  "subobject's RepNotify was called"));
+			AssertEqual_Bool(bSubobjectIntPropertyWasExpectedProperty, true,
+							 TEXT("The OnChangedRepNotifyInt1 on subobject USpatialTestRepNotifySubobject should have been set to 400 "
+								  "before the actor's RepNotify was called"));
+			FinishStep();
+		},
+		nullptr);
 }
 
 void ASpatialTestRepNotify::OnRep_OnChangedRepNotifyInt1(int32 OldOnChangedRepNotifyInt1)
 {
 	bOnRepOnChangedRepNotifyInt1Called = true;
+	bSubobjectIntPropertyWasExpectedProperty = false;
+
+	ensureAlwaysMsgf(IsValid(TestSubobject), TEXT("TestSubobject should be valid"));
+	if (TestSubobject->OnChangedRepNotifyInt == ExpectedSubobjectIntProperty)
```

Comment:
could just do `bSubobjectIntPropertyWasExpectedProperty = TestSubobject->OnChangedRepNotifyInt == ExpectedSubobjectIntProperty`, no?
But I guess it is kind of super long.

Reply:
sure, changed